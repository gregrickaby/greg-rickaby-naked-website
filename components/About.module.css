.about {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 740px;
  padding: 0 1.2rem;
}

@media (min-width: 768px) {
  .about {
    padding: 0;
  }
}
