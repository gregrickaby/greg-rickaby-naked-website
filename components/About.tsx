import styles from "./About.module.css";

export default function About() {
  return (
    <div id="about" className={styles.about}>
      <h2>About</h2>
      <p>
        Way back in 1997, I built my first website on Geocities and have been
        passionate about the web since! In my spare time I enjoy{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          taking photos
        </a>
        , cheering for the{" "}
        <a href="https://www.packers.com/">Green Bay Packers</a>, and making
        pizzas.
      </p>

      <h3>Web Developer</h3>
      <p>
        I&apos;m a Tech Lead at{" "}
        <a href="https://americaneagle.com/">AmericanEagle.com</a>, an industry
        leader in web design, development, hosting, and digital marketing. I
        partner with clients and team members to build websites with Next.js and
        WordPress.
      </p>
      <p>
        View my r√©sum√© on{" "}
        <a href="https://www.linkedin.com/in/gregrickaby/" rel="author">
          LinkedIn
        </a>
        .
      </p>

      <h3>Published Author</h3>
      <p>
        In 2017, I wrote a children&apos;s book titled{" "}
        <a href="https://amzn.to/34QAxAQ">
          <em>
            <strong>Creating a Website for Dummies Jr</strong>
          </em>
        </a>{" "}
        which was published by{" "}
        <a href="https://www.wiley.com/en-us"> Wiley Global</a> under the{" "}
        <a href="https://www.dummies.com/">dummies‚Ñ¢Ô∏è</a> brand.
      </p>

      <p>
        The book guides young readers through the planning and development
        processes of building a website. It has been translated into other 2
        other languages, and currently has a 4.5-star rating on Amazon.
      </p>

      <p>
        Since being published, I&apos;ve also served as the Technical Editor on
        three other books:
      </p>
      <ul>
        <li>
          <a href="https://amzn.to/37XCflU" rel="nofollow">
            <em>WordPress for Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2021)
        </li>
        <li>
          <a href="https://amzn.to/2BgCg7i" rel="nofollow">
            <em>Professional WordPress Plugin Development</em>
          </a>{" "}
          (Williams et al., 2020)
        </li>
        <li>
          <a href="https://amzn.to/37MMDLp" rel="nofollow">
            <em>WordPress All-In-One For Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2019)
        </li>
      </ul>

      <h3>Photographer</h3>
      <p>
        In late 2019, I bought a mirrorless camera and spent months taking
        courses, reading, practicing, and having my work critiqued by peers.
        What has started as an enjoyable hobby, has turned into a passionate way
        to express my creativity.
      </p>
      <p>
        View my{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          portfolio
        </a>{" "}
        on SmugMug.
      </p>

      <h3>Contributions</h3>
      <p>
        I've been part of the WordPress community since 2008. I've made
        contributions to core, documentation, plugins, and themes. I've also
        spoken at WordCamps and meet-ups. I'm also a contributor to other
        open-source projects like Next.js, Storybook, and Gatsby.
      </p>
      <p>
        Outside of tech, I volunteer my time and efforts at local community
        organizations. The Boy Scouts, Wiregrass Church, and the Enterprise High
        School Band Boosters to name a few.
      </p>

      <p>
        You can find me around the web with the links below. Stay safe out
        there, and thanks for reading!
        <span role="img" aria-label="cheers">
          üçª
        </span>
      </p>
    </div>
  );
}
