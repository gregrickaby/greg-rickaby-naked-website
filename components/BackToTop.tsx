'use client'

import {IconArrowUp} from '@tabler/icons-react'
import {useEffect, useState} from 'react'

/**
 * The back to top component.
 */
export default function BackToTop() {
  const [showButton, setShowButton] = useState(false)
  const buttonText = 'Go back to the top of the page'

  /**
   * Scroll handler.
   */
  const handleScroll = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth'
    })
  }

  /**
   * Effect for showing the back to top button.
   */
  useEffect(() => {
    // Scroll event handler.
    const handleScroll = () => {
      setShowButton(window.scrollY > 200)
    }

    // Add event listener.
    window.addEventListener('scroll', handleScroll)

    // Cleanup event listener.
    return () => window.removeEventListener('scroll', handleScroll)
  }, [])

  return showButton ? (
    <button
      aria-label={buttonText}
      className="button fixed bottom-8 right-3 rounded-full bg-gray-400 p-2 text-white shadow-lg hover:bg-gray-500 lg:right-12 dark:bg-gray-800 hover:dark:bg-gray-700"
      data-testid="back-to-top"
      onClick={handleScroll}
      title={buttonText}
    >
      <IconArrowUp height="32" width="32" />
      <span className="sr-only">{buttonText}</span>
    </button>
  ) : null
}
