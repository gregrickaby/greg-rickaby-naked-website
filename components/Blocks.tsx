'use client'

import '@fancyapps/ui/dist/fancybox/fancybox.css'
import {useEffect, useRef} from 'react'

/**
 * The blocks component.
 */
export default function Blocks({content}: Readonly<{content: string}>) {
  // Create a ref for the post content.
  const contentRef = useRef<HTMLDivElement>(null)

  /**
   * Initialize photo gallery.
   */
  async function galleryInit() {
    // On the server? Bail.
    if (typeof window === 'undefined') {
      return
    }

    // No ref? Bail.
    if (!contentRef.current) {
      return
    }

    // Attempt to find the gallery.
    const gallery = contentRef.current.querySelector('.grd-photo-gallery-grid')

    // No gallery? Bail.
    if (!gallery) {
      return
    }

    // Import the libraries on demand.
    const Masonry = (await import('masonry-layout')).default
    const Fancybox = (await import('@fancyapps/ui')).Fancybox
    const imagesLoaded = (await import('imagesloaded')).default

    // Initialize ImagesLoaded.
    const imagesLoadedInstance = imagesLoaded(gallery)

    // Initialize Masonry.
    const masonryGrid = new Masonry(gallery, {
      gutter: 24,
      itemSelector: '.grd-photo-gallery-image',
      columnWidth: '.grd-photo-gallery-grid-sizer',
      percentPosition: true
    })

    // When ImagesLoaded is done, initialize Masonry.
    imagesLoadedInstance.on('done', () => {
      masonryGrid?.layout?.()
    })

    // Initialize Fancybox.
    Fancybox.bind('[data-fancybox]', {
      animated: false,
      groupAll: true,
      defaultDisplay: 'flex',
      Toolbar: {
        display: {
          left: ['infobar'],
          middle: [
            'zoomIn',
            'zoomOut',
            'toggle1to1',
            'rotateCCW',
            'rotateCW',
            'flipX',
            'flipY'
          ],
          right: ['slideshow', 'thumbs', 'download', 'close']
        }
      }
    })

    // Destroy Masonry and Fancybox when the component unmounts.
    return () => {
      masonryGrid?.destroy?.()
      Fancybox.destroy?.()
    }
  }

  /**
   * Initialize the gallery when the component mounts.
   */
  useEffect(() => {
    galleryInit()
  }, [content])

  return (
    <div
      ref={contentRef}
      className="text-lg"
      dangerouslySetInnerHTML={{__html: content}}
    />
  )
}
