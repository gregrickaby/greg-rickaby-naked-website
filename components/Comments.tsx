import CommentForm from '@/components/CommentForm'
import type {Post} from '@/lib/types'

/**
 * Comments component.
 */
export default function Comments({post}: Readonly<{post: Post}>) {
  return (
    <>
      <section className="border-t-2 border-zinc-500 py-4">
        <h3 className="text-3xl font-bold">Comments</h3>
        {!!post?.comments?.edges?.length && (
          <p className="italic">No comments yet.</p>
        )}
        {post.comments.edges.map((comment) => (
          <article key={comment.node.databaseId}>
            <header className="flex items-center gap-2">
              {!!comment?.node?.author?.node?.avatar?.url && (
                <img
                  alt={comment.node.author.node.name}
                  className="m-0 rounded-full"
                  height={64}
                  src={comment.node.author.node.avatar.url}
                  width={64}
                />
              )}
              <div className="flex flex-col gap-2">
                <h4
                  className="m-0 p-0 leading-none"
                  dangerouslySetInnerHTML={{
                    __html: comment.node.author.node.name
                  }}
                />
                <time className="italic">{comment.node.date}</time>
              </div>
            </header>
            <div dangerouslySetInnerHTML={{__html: comment.node.content}} />
          </article>
        ))}
      </section>
      <CommentForm postID={post.databaseId} />
    </>
  )
}
