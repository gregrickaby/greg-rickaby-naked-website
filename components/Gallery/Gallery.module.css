.gallery {
  @apply columns-1 gap-5 sm:columns-2 sm:gap-8 md:columns-3 lg:columns-4 [&>img:not(:first-child)]:mt-8;
}

.photo {
  @apply cursor-pointer;
}
