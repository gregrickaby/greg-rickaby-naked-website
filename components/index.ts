export * from './ArticleCard'
export * from './BackToTop'
export * from './Blocks'
export * from './BlogArchive'
export * from './Byline'
export * from './CommentForm'
export * from './Comments'
export * from './Footer'
export * from './Header'
export * from './JsonLd'
export * from './Navigation'
export * from './SocialLinks'
