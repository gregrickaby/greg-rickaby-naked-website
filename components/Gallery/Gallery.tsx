'use client'

import {Lightbox} from '@/components/Lightbox'
import {Threads} from '@/lib/api'
import {useCallback, useState} from 'react'
import styles from './Gallery.module.css'

export interface GalleryProps {
  photos: Threads
}

/**
 * Photos component.
 */
export function Gallery({photos}: GalleryProps) {
  // Manage whether the lightbox is open.
  const [isOpen, setIsOpen] = useState(false)
  // Manage the index of the currently selected photo.
  const [photoIndex, setPhotoIndex] = useState(0)

  // Open the lightbox and set the current photo index.
  const openLightbox = useCallback((index: number) => {
    setPhotoIndex(index)
    setIsOpen(true)
  }, [])

  // Close the lightbox.
  const closeLightbox = useCallback(() => {
    setIsOpen(false)
  }, [])

  // No photos? Don't render.
  if (!photos || photos.data.length === 0) {
    return null
  }

  return (
    <div className={styles.gallery} data-testid="gallery">
      {photos.data.map((photo, index) => (
        <div
          aria-label={`Open photo ${photo.text}`}
          className={styles.photo}
          data-testid={`photo-${index}`}
          key={photo.id}
          onClick={() => openLightbox(index)}
          onKeyPress={(e) => {
            if (e.key === 'Enter') openLightbox(index)
          }}
          role="button"
          tabIndex={0}
        >
          <img
            alt={photo.text}
            height={300}
            loading="lazy"
            src={photo.media_url}
            width={425}
          />
        </div>
      ))}

      {isOpen && (
        <Lightbox
          photos={photos}
          onClose={closeLightbox}
          startIndex={photoIndex}
        />
      )}
    </div>
  )
}
