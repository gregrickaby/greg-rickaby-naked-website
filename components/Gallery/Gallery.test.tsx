import {Gallery} from '@/components/Gallery'
import {mockEmptyThreads, mockThreads} from '@/mocks/mockThreads'
import '@testing-library/jest-dom'
import {render, waitFor} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {axe} from 'jest-axe'
import {describe, expect, it} from 'vitest'

describe('Gallery', () => {
  it('should render', () => {
    const {getByTestId} = render(<Gallery photos={mockThreads} />)
    const gallery = getByTestId('gallery')
    expect(gallery).toBeInTheDocument()
  })

  it('should not render if there are no photos', () => {
    const {queryByTestId} = render(<Gallery photos={mockEmptyThreads} />)
    const gallery = queryByTestId('gallery')
    expect(gallery).not.toBeInTheDocument()
  })

  it('should open the lightbox when a photo is clicked', async () => {
    const {getByTestId} = render(<Gallery photos={mockThreads} />)
    const photo = getByTestId('photo-0')

    await userEvent.click(photo)

    waitFor(() => {
      const lightbox = getByTestId('lightbox')
      expect(lightbox).toBeInTheDocument()
    })
  })

  it('should open the lightbox by pressing Enter on a photo', async () => {
    const {getByTestId} = render(<Gallery photos={mockThreads} />)
    const photo = getByTestId('photo-0')

    await userEvent.type(photo, '{enter}')

    waitFor(() => {
      const lightbox = getByTestId('lightbox')
      expect(lightbox).toBeInTheDocument()
    })
  })

  it('should close the lightbox when the close button is clicked', async () => {
    const {getByTestId} = render(<Gallery photos={mockThreads} />)
    const photo = getByTestId('photo-0')

    await userEvent.click(photo)

    waitFor(() => {
      const lightbox = getByTestId('lightbox')
      expect(lightbox).toBeInTheDocument()
    })

    const closeButton = getByTestId('close-button')
    await userEvent.click(closeButton)

    waitFor(() => {
      const lightbox = getByTestId('lightbox')
      expect(lightbox).not.toBeInTheDocument()
    })
  })

  it('should not have any accessibility issues', async () => {
    const {container} = render(<Gallery photos={mockThreads} />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it('should match snapshot', () => {
    const {getByTestId} = render(<Gallery photos={mockThreads} />)
    const gallery = getByTestId('gallery')
    expect(gallery).toMatchSnapshot()
  })
})
