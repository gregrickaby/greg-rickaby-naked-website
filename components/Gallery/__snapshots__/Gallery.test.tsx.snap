// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gallery > should match snapshot 1`] = `
<div
  class="_gallery_5fdb3a"
  data-testid="gallery"
>
  <div
    aria-label="Open photo Taking advantage of the new moon! Milky Way from my front yard"
    class="_photo_5fdb3a"
    data-testid="photo-0"
    role="button"
    tabindex="0"
  >
    <img
      alt="Taking advantage of the new moon! Milky Way from my front yard"
      height="300"
      loading="lazy"
      src="https://scontent-atl3-1.cdninstagram.com/v/t51.29350-15/453769781_1165705561342861_3077480464820431675_n.jpg?_nc_cat=107&ccb=1-7&_nc_sid=18de74&_nc_ohc=nRWYYqypj0oQ7kNvgErE_Lc&_nc_ht=scontent-atl3-1.cdninstagram.com&edm=ACx9VUEEAAAA&oh=00_AYBVVGC6uLoGHbwqC5kMNBcEHVYPpneWytiQ5AC-BW6y2w&oe=66B3F393"
      width="425"
    />
  </div>
  <div
    aria-label="Open photo My youngest yelled, “it’s the zombie apocalypse!” stormyweather"
    class="_photo_5fdb3a"
    data-testid="photo-1"
    role="button"
    tabindex="0"
  >
    <img
      alt="My youngest yelled, “it’s the zombie apocalypse!” stormyweather"
      height="300"
      loading="lazy"
      src="https://scontent-atl3-1.cdninstagram.com/v/t51.29350-15/453259251_503995798667592_3625227248165730293_n.jpg?_nc_cat=109&ccb=1-7&_nc_sid=18de74&_nc_ohc=46uSKISPRVkQ7kNvgEY2ddI&_nc_ht=scontent-atl3-1.cdninstagram.com&edm=ACx9VUEEAAAA&oh=00_AYDm265xezWddW7GUpVbRMwgEbMmNzvJyuskddCXhcsuTw&oe=66B3CB08"
      width="425"
    />
  </div>
  <div
    aria-label="Open photo Who wants hugs?"
    class="_photo_5fdb3a"
    data-testid="photo-2"
    role="button"
    tabindex="0"
  >
    <img
      alt="Who wants hugs?"
      height="300"
      loading="lazy"
      src="https://scontent-atl3-1.cdninstagram.com/v/t51.29350-15/452937348_526651603050710_3094736639257777462_n.jpg?_nc_cat=106&ccb=1-7&_nc_sid=18de74&_nc_ohc=KVJHOvmacecQ7kNvgHYSpNQ&_nc_ht=scontent-atl3-1.cdninstagram.com&edm=ACx9VUEEAAAA&oh=00_AYBXIiM3xGOzCqEoyvBh8z-_DDhLfGSAGy5YtZKTeg4BZg&oe=66B3CC28"
      width="425"
    />
  </div>
  <div
    aria-label="Open photo Hello dark sky season! 👀🌌📷"
    class="_photo_5fdb3a"
    data-testid="photo-3"
    role="button"
    tabindex="0"
  >
    <img
      alt="Hello dark sky season! 👀🌌📷"
      height="300"
      loading="lazy"
      src="https://scontent-atl3-1.cdninstagram.com/v/t39.30808-6/449195864_18439774633004645_5040392547010828344_n.jpg?_nc_cat=100&ccb=1-7&_nc_sid=18de74&_nc_ohc=6YpL6AguGZAQ7kNvgE3kRKJ&_nc_ht=scontent-atl3-1.cdninstagram.com&edm=ACx9VUEEAAAA&oh=00_AYDdRd50T67gEVWGXoKS5IDit4QWkKfeBoXRaXhUzvxyyw&oe=66B3F128"
      width="425"
    />
  </div>
  <div
    aria-label="Open photo Happy Birthday @chefboyarecee!!! 1️⃣9️⃣"
    class="_photo_5fdb3a"
    data-testid="photo-4"
    role="button"
    tabindex="0"
  >
    <img
      alt="Happy Birthday @chefboyarecee!!! 1️⃣9️⃣"
      height="300"
      loading="lazy"
      src="https://scontent-atl3-1.cdninstagram.com/v/t39.30808-6/448399927_18437829799004645_8126595902947242612_n.jpg?_nc_cat=108&ccb=1-7&_nc_sid=18de74&_nc_ohc=raCixRpmvRUQ7kNvgE-wQjj&_nc_ht=scontent-atl3-1.cdninstagram.com&edm=ACx9VUEEAAAA&oh=00_AYBpOpkg0bqotEfIYBq4Y0bEr9PSk4cI2fckDnT-5L8Vng&oe=66B3E93B"
      width="425"
    />
  </div>
</div>
`;
